doc/sphinx/source/utils/RTW/add_a_recipe.rst

```diff
@@ -0,0 +1,112 @@
+
+     YYYY-MM-DD HH:MM:SS:sss UTC [12345] INFO    Time for running the recipe was: 0:02:13.334742
+     YYYY-MM-DD HH:MM:SS:sss UTC [12345] INFO    Maximum memory used (estimate): 2.4 GB
+     [...]
+     YYYY-MM-DD HH:MM:SS:sss UTC [12345] INFO    Run was successful
+
+#. Add the recipe to the ``[task parameters]`` section in the
+   ``esmvaltool/utils/recipe_test_workflow/flow.cylc`` file.
+
+   .. hint::
+      If the recipe takes less than 10 minutes to run then it should be added
+      to the ``fast`` option. Recipes that take longer than ten minutes should
+      be added to the ``medium`` option.
+
+   .. hint::
+      The line added should follow the format of ``recipe_new_recipe, \``,
+      unless the line is the last one in the list, in which case the line added
+      should follow the format of ``recipe_new_recipe``.
+
+#. If the duration of the recipe is larger than the value specified by the
+   ``execution time limit`` option in the ``[[COMPUTE]]`` section in the
+   aforementioned site-specific ``.cylc`` file, and / or the memory usage of
+   the recipe is larger than the value specified by the ``--mem`` option in the
+   ``[[[directives]]]`` section in the ``[[COMPUTE]]`` section, add a section
+   (in alphabetical order) to this file as shown below (round the duration to
+   the nearest second)::
+
+     [[process<fast=recipe_albedolandcover>]]
+     # Actual: 0m31s, 2.5 GB on 2024-04-08.
+     execution time limit = PT2M
+     [[[directives]]]
+         --mem = 3G
+
+   .. hint::
+      The ``fast`` key in the example task definition above
+      (``[[process<fast=recipe_albedolandcover>]]``) should match name of the
+      option the recipe was added to in the ``[task parameters]`` section in
+      the ``esmvaltool/utils/recipe_test_workflow/flow.cylc`` file
+
+   .. hint::
+      Set the ``execution time limit`` to 10-20% more than the actual duration.
+      For actual durations of up to ``1m45s``, set the ``execution time limit``
+      to ``PT2M`` (2 minutes).
+
+   .. hint::
+      Try not to regularly waste more than 500 MiB in memory usage. Typically,
+      rounding the actual memory usage up to the nearest integer is acceptable.
+
+#. Stop any running ``recipe_test_workflow`` workflows::
+
+    cylc stop recipe_test_workflow/*
+
+#. Run the |RTW|, as detailed in the :ref:`quick_start_guide`; it is expected
+   that the ``compare`` task will fail.
+
+#. Update the Known Good Outputs (|KGOs|):
+
+   * Recursively copy the recipe output directory (i.e.
+     ``recipe_<recipe>_<date>_<time>/``) from the
+     ``${HOME}/cylc-run/recipe_test_workflow/runN/share/cycle/<cycle>/``
+     directory to your site-specific |KGO| directory, as detailed by the
+     ``KGO_ROOT_PATH`` option in the site-specific ``.conf`` file in the
+     ``esmvaltool/utils/recipe_test_workflow/opt/`` directory::
+
+       cp -r ${HOME}/cylc-run/recipe_test_workflow/runN/share/cycle/<cycle>/recipe_<recipe>_<date>_<time> <KGO_ROOT_PATH>
```

Comment:
Shouldn't the output of the recipes be found in another directory rather than the HOME one? Maybe generalise it to the `esmvaltool_output` directory ? At least in Levante you would run out space pretty quickly if you store the recipe output in the home.

Reply:
Ah, there is some Cylc magic going on here; I will add a note here to say that `${HOME}/cylc-run/recipe_test_workflow/runN/share` is a symbolic link to e.g. on DKRZ: `/scratch/b/b382148/cylc-run/recipe_test_workflow/runN/share`. Large files are not stored in `${HOME}` :blush:

Reply:
Change made: 284b66d